### ISSUE TITLE
Pytest failures due to duplicated test file names

### ISSUE THREAD

**Issue by busunkim96** [googleapis/gapic-generator-python#263]
I generated text-to-speech, which has two versions (v1 and v1beta1). The `tests` directory looks like this:

```
tests
└── unit
    ├── texttospeech_v1
    │   └── test_text_to_speech.py
    └── texttospeech_v1beta1
        └── test_text_to_speech.py
```

The tests run fine with the noxfile provided by the generator [`--cov={{ api.naming.module_namespace|join("/") }}/{{ api.naming.versioned_module_name }}/`](https://github.com/googleapis/gapic-generator-python/blob/ccfd2d34161f74d14765e1e9b7e398bf14f4aa89/gapic/templates/noxfile.py.j2#L16-L24), but fail with a less specific invocation like `--cov=tests.unit`.
```
___________ ERROR collecting tests/unit/texttospeech_v1beta1/test_text_to_speech.py ____________
import file mismatch:
imported module 'test_text_to_speech' has this __file__ attribute:
  /usr/local/google/home/busunkim/github/google-cloud-python/texttospeech/tests/unit/texttospeech_v1/test_text_to_speech.py
which is not the same as the test file we want to collect:
  /usr/local/google/home/busunkim/github/google-cloud-python/texttospeech/tests/unit/texttospeech_v1beta1/test_text_to_speech.py
HINT: remove __pycache__ / .pyc files and/or use a unique basename for your test file modules
```

According to https://github.com/pytest-dev/pytest/issues/3151, every test file name should be distinct name or there should be `__init__.py` files. The monolith seems to go the route of having distinct names for every file.

```
tests
├── system
│   └── gapic
│       ├── v1
│       │   └── test_system_speech_v1.py
│       └── v1p1beta1
│           └── test_system_speech_v1p1beta1.py
└── unit
    ├── gapic
    │   ├── v1
    │   │   └── test_speech_client_v1.py
    │   └── v1p1beta1
    │       └── test_speech_client_v1p1beta1.py
    └── test_helpers.py
```

**Comment by software-dov** [comment]
Leaning towards having an `__init__.py` file unless you feel strongly otherwise.
Mild soapbox: names should be short and pithy, paths should be detailed and descriptive.

**Comment by busunkim96** [comment]
@software-dov That sounds good to me.

**Comment by busunkim96** [comment]
Closed by #270
